inject/src/main/java/io/avaje/inject/RequiresBean.java

```diff
@@ -0,0 +1,56 @@
+package io.avaje.inject;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Expresses a requirement for a bean to be wired/registered into the {@link BeanScope}.
+ *
+ * <pre class="code">
+ * &#064;Configuration
+ * public class MyAutoConfiguration {
+ *
+ *     &#064;Bean
+ *     &#064;Requires(beans = OtherService.class)
+ *     public MyService myService() {
+ *         ...
+ *     }
+ *
+ * }</pre>
+ *
+ * <p>In the sample above the MyService bean will get wired only if a bean of type {@code
+ * OtherService} is already registered in the {@link BeanScope}.
+ */
+@Retention(RUNTIME)
+@Repeatable(RequireBeans.class)
+@Target({TYPE, METHOD, ANNOTATION_TYPE})
+public @interface RequiresBean {
+
+  /**
+   * Expresses that beans of the given types should be available in the {@link BeanScope}.
+   *
+   * @return the class types of beans to check
+   */
+  Class<?>[] value() default {};
```

Comment:
perhaps, but the reason I took that out is that I thought there would be some inconsistency with how the modules get ordered, since unlike the Class example we can't define a soft dependency and auto require on a class not in the cp